app/models/student_fields.rb

```diff
@@ -0,0 +1,34 @@
+module StudentFields
```

Comment:
"Concern" I have with this: we're polluting the already **huge** `models` folder with things that aren't models and only make sense when combined with an actual *model*. Therefore I'd vote for: making it a `concern` 😂 

There's the `concerns` directory in the `models` directory. You could add a fitting namespace, maybe `Fields` and then stuff it there:
```ruby
# in app/models/concerns/fields/student_fields.rb

module Fields
  module StudentFields
    # all the fun
```

What do you think?

Reply:
I was in doubt about it, because I learned not to use `concerns` for things that are in one model only. But it looks so much better, so  👍   totally agreed.

Reply:
On one hand, I agree that this module shouldn't reside in app/models. Otoh, why do we care? There is no real definition what should go into models. Only persistence objects? We would have to through out quite a few files then. And were to put them? Coming up with new subfolders under `app/` makes it more difficult to find the implementation of a class than browsing a densely  packed app/models folder.

The current implementation of `StudentFields` does look as if it belonged into `app/concerns`.

That being said, I'm not so much a fan of models (concerns) as a bandaid to remove the line count of a fat model when they're only used in a single class. If anything, it makes it only less explicit what a model is capable of.

If we don't want to expose constants to the form, can we transform these to module functions? Or, better yet (and certainly not in this PR), add them as methods to a custom form builder for the the students fields form?

Reply:
We can decide later on a more permanent decision? I'll keep all this in mind, together with the option of decorators.

Reply:
👍 

I think the important part was just to realize that these things should be part of the form object rather then the user model - and we have that now.